### PR TITLE
[fix][admin] Add javax.xml.bind:jaxb-api to shade

### LINKED ISSUE #19856: Update jaxb and jax-rs libs
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Motivation

Is there any plan to update Jaxb and JaxRS libs (from javax.* to jakarta.*)? 

When using the Pulsar admin client `2.11.0` in a modern application that is using the more recent versions of Jaxb and JaxRS, the recent versions are the ones used. This may lead to confusion. The only reason it is working in this case is because the elder` jaxb-api` is still carried along that has the old annotations. 

### Solution

BUMP jakarta.ws.rs:jakarta.ws.rs-api from 2.1.6	to 3.1.0
BUMP jakarta.xml.bind:jakarta.xml.bind-api from	2.3.3 4.0.0
Remove javax.xml.bind:jaxb-api:2.3.1

Update all code references from javax.* to jakarta.*

### Alternatives

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by nodece**
Fixes #19856

### Motivation

`javax.xml.bind:jaxb-api` didn't package into the shade package.

Add a context, why add `javax.xml.bind:jaxb-api` to the admin module, see https://www.jesperdj.com/2018/09/30/jaxb-on-java-9-10-11-and-beyond/

### Modifications

Add `javax.xml.bind:jaxb-api` to  `pulsar-client-admin-shaded/pom.xml`.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

File: pulsar-client-admin-shaded/pom.xml
Comment (onobc) [inline pulsar-client-admin-shaded/pom.xml:134]: @nodece this inclusion/relocation looks good. I think we need to do the same thing to [pulsar-client-all/pom.xml](https://github.com/apache/pulsar/blob/091ee2504ffbe6ec98e354b76e7f4c045e1914aa/pulsar-client-all/pom.xml#L184).

Comment (nodece) [reply pulsar-client-admin-shaded/pom.xml:134]: I'll make a new PR to do that.

Comment (nodece) [reply pulsar-client-admin-shaded/pom.xml:134]: @onobc Done.

**Review by merlimat — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]
Thank you!

**Comment by onobc** [comment]
Is there test coverage for a case where the admin API `PulsarAdmin.topics().getList("public/default")` is executed against a standalone server? The reason I ask is because the `2.11.1` branch w/o the changes in this PR are fine but w/ these changes in the PR a call to the above API just times out w/ the following output in the logs:

> 2023-04-15T16:11:43.776-05:00  WARN 4356 --- [           main] o.a.p.s.o.g.j.m.i.MessagingBinders       : A class org.apache.pulsar.shade.javax.xml.transform.stream.StreamSource for a default provider MessageBodyReader<org.apache.pulsar.shade.javax.xml.transform.stream.StreamSource> was not found. The provider is not available.
2023-04-15T16:11:43.777-05:00  WARN 4356 --- [           main] o.a.p.s.o.g.j.m.i.MessagingBinders       : A class org.apache.pulsar.shade.javax.xml.transform.sax.SAXSource for a default provider MessageBodyReader<org.apache.pulsar.shade.javax.xml.transform.sax.SAXSource> was not found. The provider is not available.
2023-04-15T16:11:43.777-05:00  WARN 4356 --- [           main] o.a.p.s.o.g.j.m.i.MessagingBinders       : A class org.apache.pulsar.shade.javax.xml.transform.Source for a default provider MessageBodyWriter<org.apache.pulsar.shade.javax.xml.transform.Source> was not found. The provider is not available.
2023-04-15T16:11:43.777-05:00  WARN 4356 --- [           main] o.a.p.s.o.g.j.m.i.MessagingBinders       : A class javax.activation.DataSource for a default provider MessageBodyWriter<javax.activation.DataSource> was not found. The provider is not available.
2023-04-15T16:11:43.777-05:00  WARN 4356 --- [           main] o.a.p.s.o.g.j.m.i.MessagingBinders       : A class org.apache.pulsar.shade.javax.xml.transform.dom.DOMSource for a default provider MessageBodyReader<org.apache.pulsar.shade.javax.xml.transform.dom.DOMSource> was not found. The provider is not available.

I am concerned that shading the `javax.xml.bind` may need updates in other areas? If someone else could verify that they are NOT seeing this behaviour w/ these changes that would be comforting.

Thanks

**Comment by onobc** [comment]
> I am concerned that shading the `javax.xml.bind` may need updates in other areas? If someone else could verify that they are NOT seeing this behaviour w/ these changes that would be comforting.
> 
> Thanks

I believe the current relocation is too aggressive as it attempts to relocate JDK classes (`javax.xml.DOMSource`)  in `MessageBinders` but will not shade those classes actually because they are JDK classes. 

I got further by tightening the shading to `javax.xml.bind` w/ 
```
 <include>javax.xml.bind:jaxb-api</include>
```
instead of 
```
<include>javax.xml.bind:*</include>
```
and for relocation:
```
<relocation>
  <pattern>javax.xml.bind</pattern>
  <shadedPattern>org.apache.pulsar.shade.javax.xml.bind</shadedPattern>
</relocation>
```
instead of 
```
<relocation>
  <pattern>javax.xml</pattern>
  <shadedPattern>org.apache.pulsar.shade.javax.xml</shadedPattern>
</relocation>
```
The warnings now go away in the app log and the call does not timeout. However, another error now occurs. I am debugging and hopefully will have some info soon.

**Comment by nodece** [comment]
Hi @onobc, thank you for your research! This PR has been updated.

**Comment by tisonkun** [comment]
Test failure can be related:

```
Error:  Tests run: 18, Failures: 1, Errors: 0, Skipped: 2, Time elapsed: 55.518 s <<< FAILURE! - in org.apache.pulsar.broker.loadbalance.extensions.ExtensibleLoadManagerImplTest
  Error:  initializeState(org.apache.pulsar.broker.loadbalance.extensions.ExtensibleLoadManagerImplTest)  Time elapsed: 34.049 s  <<< FAILURE!
  org.apache.pulsar.client.admin.PulsarAdminException$ServerSideErrorException: HTTP 500 {}
  	at org.apache.pulsar.client.admin.PulsarAdminException.wrap(PulsarAdminException.java:252)
  	at org.apache.pulsar.client.admin.internal.BaseResource.sync(BaseResource.java:352)
  	at org.apache.pulsar.client.admin.internal.NamespacesImpl.unload(NamespacesImpl.java:792)
  	at org.apache.pulsar.broker.loadbalance.extensions.ExtensibleLoadManagerImplTest.initializeState(ExtensibleLoadManagerImplTest.java:173)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
  	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
  	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethodConsideringTimeout(MethodInvocationHelper.java:69)
  	at org.testng.internal.invokers.ConfigInvoker.invokeConfigurationMethod(ConfigInvoker.java:361)
  	at org.testng.internal.invokers.ConfigInvoker.invokeConfigurations(ConfigInvoker.java:296)
  	at org.testng.internal.invokers.TestInvoker.runConfigMethods(TestInvoker.java:823)
  	at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:590)
  	at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:221)
  	at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
  	at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:969)
  	at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:194)
  	at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
  	at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
  	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
  	at org.testng.TestRunner.privateRun(TestRunner.java:829)
  	at org.testng.TestRunner.run(TestRunner.java:602)
  	at org.testng.SuiteRunner.runTest(SuiteRunner.java:437)
  	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:431)
  	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:391)
  	at org.testng.SuiteRunner.run(SuiteRunner.java:330)
  	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
  	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
  	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1256)
  	at org.testng.TestNG.runSuitesLocally(TestNG.java:1176)
  	at org.testng.TestNG.runSuites(TestNG.java:1099)
  	at org.testng.TestNG.run(TestNG.java:1067)
  	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:135)
  	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeSingleClass(TestNGDirectoryTestSuite.java:112)
  	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeLazy(TestNGDirectoryTestSuite.java:123)
  	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:90)
  	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:146)
  	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
  	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
  	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
  	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
  	Suppressed: org.apache.pulsar.client.admin.PulsarAdminException$ServerSideErrorException: HTTP 500 {}
  		at org.apache.pulsar.client.admin.internal.BaseResource.getApiException(BaseResource.java:272)
  		at org.apache.pulsar.client.admin.internal.BaseResource$1.failed(BaseResource.java:136)
  		at org.glassfish.jersey.client.JerseyInvocation$1.failed(JerseyInvocation.java:882)
  		at org.glassfish.jersey.client.JerseyInvocation$1.completed(JerseyInvocation.java:863)
  		at org.glassfish.jersey.client.ClientRuntime.processResponse(ClientRuntime.java:229)
  		at org.glassfish.jersey.client.ClientRuntime.access$200(ClientRuntime.java:62)
  		at org.glassfish.jersey.client.ClientRuntime$2.lambda$response$0(ClientRuntime.java:173)
  		at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
  		at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
  		at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
  		at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
  		at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
  		at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:288)
  		at org.glassfish.jersey.client.ClientRuntime$2.response(ClientRuntime.java:173)
  		at org.apache.pulsar.client.admin.internal.http.AsyncHttpConnector.lambda$apply$1(AsyncHttpConnector.java:254)
  		at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
  		at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
  		at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
  		at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
  		at org.apache.pulsar.client.admin.internal.http.AsyncHttpConnector.lambda$retryOperation$4(AsyncHttpConnector.java:296)
  		at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
  		at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
  		at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
  		at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
  		at org.asynchttpclient.netty.NettyResponseFuture.loadContent(NettyResponseFuture.java:222)
  		at org.asynchttpclient.netty.NettyResponseFuture.done(NettyResponseFuture.java:257)
  		at org.asynchttpclient.netty.handler.AsyncHttpClientHandler.finishUpdate(AsyncHttpClientHandler.java:241)
  		at org.asynchttpclient.netty.handler.HttpHandler.handleChunk(HttpHandler.java:114)
  		at org.asynchttpclient.netty.handler.HttpHandler.handleRead(HttpHandler.java:143)
  		at org.asynchttpclient.netty.handler.AsyncHttpClientHandler.channelRead(AsyncHttpClientHandler.java:78)
  		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
  		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
  		at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
  		at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
  		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
  		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
  		at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
  		at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436)
  		at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346)
  		at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318)
  		at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251)
  		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
  		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
  		at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
  		at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
  		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
  		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
  		at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
  		at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
  		at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
  		at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
  		at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
  		at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
  		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
  		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
  		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
  		at java.base/java.lang.Thread.run(Thread.java:833)
  	Caused by: javax.ws.rs.InternalServerErrorException: HTTP 500 {}
  		at org.glassfish.jersey.client.JerseyInvocation.convertToException(JerseyInvocation.java:960)
  		at org.glassfish.jersey.client.JerseyInvocation.access$[700](https://github.com/apache/pulsar/actions/runs/4712460077/jobs/8357595863#step:11:701)(JerseyInvocation.java:82)
  		... 54 more
  Caused by: [CIRCULAR REFERENCE: javax.ws.rs.InternalServerErrorException: HTTP 500 {}]
```

**Comment by cbornet** [comment]
@merlimat we are normally in code freeze for 3.0. Do we do an exception for this one ?

**Comment by onobc** [comment]
FWIW - this change is one that is required for Spring Pulsar to get it's auto-configuration moved into Spring Boot (3.2) project. Thanks.

**Comment by merlimat** [comment]
@cbornet Yes, this a very safe and understood change, that would not add more risk to the RC-2.

**Comment by onobc** [comment]
I don't see this one in the [3.0.0 Release notes](https://github.com/apache/pulsar/releases/tag/v3.0.0) but I do see the changes in the [branch-3.0](https://github.com/apache/pulsar/blob/7636e8989f4d3fc24fce69a356d54e1c550945ed/pulsar-client-all/pom.xml#L185). Really I just want to make sure the fix is in 3.0.0 and there is not something else I am missing. 

Thanks

**Comment by tisonkun** [comment]
cc @RobertIndie perhaps you can investigate why the release note is unsync?

**Comment by RobertIndie** [comment]
The release note is collected by filtering the label like `component/admin`. This PR is not labeled with `component/admin`. So it's not included in the release note. 

It's hard for the release manager to collect and classify all the unlabeled PRs. I encourage we should label all the PRs correctly before merging the PR.

**Comment by tisonkun** [comment]
Cross-reference - https://github.com/apache/flink-connector-pulsar/pull/54

This include a new dependency rejected by Flink's LicenseChecker, while it should be a false positive.

**Pull request merged**
